device/device.go

```diff
@@ -12,17 +12,19 @@ import (
 )
 
 type Device struct {
-	path         string
-	file         *os.File
-	fd           uintptr
-	config       config
-	bufType      v4l2.BufType
-	cap          v4l2.Capability
-	cropCap      v4l2.CropCapability
-	buffers      [][]byte
-	requestedBuf v4l2.RequestBuffers
-	streaming    bool
-	output       chan []byte
+	path          string
+	file          *os.File
+	fd            uintptr
+	config        config
+	bufType       v4l2.BufType
+	cap           v4l2.Capability
+	cropCap       v4l2.CropCapability
+	buffers       [][]byte
+	requestedBuf  v4l2.RequestBuffers
+	streaming     bool
+	output        chan []byte
+	context       context.Context
```

Comment:
I don't think the context needs to be stored in this struct.  There's some good documentation about this in the Go blog.

> Contexts should not be stored inside a struct type, but instead passed to each function that needs it.

https://go.dev/blog/context-and-structs